File: crate/fixtures/documents/patterns.go
package main

import "regexp"

var (
	// A backquoted string is the source verbatim
	nested = regexp.MustCompile(`(a+)+`)
	// A quoted string escapes a level
	escaped = regexp.Compile("\\d{2}:\\d{2}")
	// Inline flags: valid RE2, and not JavaScript
	inline = regexp.MustCompile(`(?i)^[a-z]+$`)
)

File: crate/fixtures/documents/patterns.py
import re

# The textbook catastrophic-backtracking shape, in a raw string
NESTED = re.compile(r"(a+)+")

# A quoted string escapes a level, exactly as the JS constructor does
ESCAPED = re.compile("\\d{3}-\\d{4}")

# Single-quoted raw, and a triple-quoted body
SINGLE = re.match(r'^[a-z]+$', name)
TRIPLE = re.search("""(a|ab)+""", text)

# A path is not a pattern: Python has no /…/ literal
PREFIX = "/usr/local/bin"
